### PR TITLE
enable Azure appinsights/tracing

### PR DESCRIPTION
**Pull request by vrajroutu**
## Purpose

* ...

## Does this introduce a breaking change?

```
[ ] Yes
[ X] No
```

## Pull Request Type
What kind of change does this Pull Request introduce?

```
[ ] Bugfix
[ X] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Documentation content changes
[ ] Other... Please describe:
```

## How to Test
*  Get the code

```
git clone [repo-address]
cd [repo-name]
git checkout [branch-name]
npm install
```

* Test the code

```
```

## What to Check
Verify that the following are valid
* ...

### CODE REVIEW

**Comment by kishorerv93** [comment]
@pamelafox  can you review this PR?

**Comment by kishorerv93** [comment]
Screenshot 2023-07-11 at 11 38 20 PM

File: app/backend/app.py [hunk 1]
```diff
@@ -75,50 +95,53 @@ def static_file(path):
...
-    try:
-        impl = chat_approaches.get(approach)
-        if not impl:
-            return jsonify({"error": "unknown approach"}), 400
-        r = impl.run(request.json["history"], request.json.get("overrides") or {})
-        return jsonify(r)
-    except Exception as e:
-        logging.exception("Exception in /chat")
-        return jsonify({"error": str(e)}), 500
+        abort(404)
```
Comment (pamelafox) [inline app/backend/app.py:107]: can you avoid the whitespace changes on this diff? (i think all these lines are just whitespace changes)

File: app/backend/app.py [hunk 2]
```diff
@@ -65,6 +72,19 @@
...
+tracer = trace.get_tracer(__name__)  
+  
+# Set up Azure Monitor Trace Exporter  
+azure_exporter = AzureMonitorTraceExporter.from_connection_string(AZURE_MONITOR_CONNECTION_STRING)  
+span_processor = BatchSpanProcessor(azure_exporter)  
+trace.get_tracer_provider().add_span_processor(span_processor) 
+
+# Set up logging  
+logging.basicConfig(level=logging.INFO)  
+logger = logging.getLogger(__name__)
```
Comment (pamelafox) [inline app/backend/app.py:86]: Is the logger variable referenced somewhere here? In Flask, its usually recommended to use app.logger:
https://flask.palletsprojects.com/en/2.3.x/logging/

File: app/backend/approaches/chatreadretrieveread.py [hunk 3]
```diff
@@ -53,61 +60,71 @@ def __init__(self, search_client: SearchClient, chatgpt_deployment: str, chatgpt
...
-            model=self.chatgpt_model,
-            messages=messages, 
-            temperature=overrides.get("temperature") or 0.7, 
-            max_tokens=1024, 
-            n=1)
-        
-        chat_content = chat_completion.choices[0].message.content
-
-        msg_to_display = '\n\n'.join([str(message) for message in messages])
+        # STEP 1: Generate an optimized keyword search query based on the chat history and the last question
```
Comment (pamelafox) [inline app/backend/approaches/chatreadretrieveread.py:76]: Whitespace changes?

File: app/backend/approaches/chatreadretrieveread.py [hunk 4]
```diff
@@ -6,6 +6,13 @@
 from azure.search.documents.models import QueryType
 from approaches.approach import Approach
 from text import nonewlines
+from opencensus.ext.azure.log_exporter import AzureLogHandler
```
Comment (pamelafox) [inline app/backend/approaches/chatreadretrieveread.py:9]: Why does this file use opencensus and the other one uses opentelemetry?

**Review by pamelafox — CHANGES_REQUESTED** [review]
Please fill out the PR with all the fields, like description.

**Comment by pamelafox** [comment]
@vrajroutu I see you closed this - are you planning to re-open or make a new PR? I do think AppInsights would be great to support.

**Comment by vrajroutu** [comment]
@pamelafox; Yes i'm working on submitting a new PR. will be there in a bit :)

**Pull request closed without merging**